paramiko/agent.py

```diff
@@ -109,9 +109,12 @@ def __init__(self, agent):
     def run(self):
         try:
             (r, addr) = self.get_connection()
+            # Found that r should be either a socket from the socket library or None
             self.__inr = r
-            self.__addr = addr
+            self.__addr = addr # This should be an IP address as a string? or None
             self._agent.connect()
+            if self._agent._conn is None or not (hasattr(self._agent._conn, 'fileno') or isinstance(self._agent, int)):
```

Comment:
Probably need `isinstance` first, otherwise if it *is* an int, the first test(s) on this line will fail since `int` types do not have `._conn` attributes :)

I actually don't recall seeing reports of an issue where `self._agent` is an int - where did this come from?